Thunder/utils/custom_dl.py

```
@@ -1,371 +1,60 @@
-        file_id: FileId,
-        index: int,
-        offset: int,
-        first_part_cut: int,
-        last_part_cut: int,
-        part_count: int,
-        chunk_size: int,
-    ) -> AsyncGenerator[bytes, None]:
-        client = self.client
-        work_loads[index] += 1
+            logger.debug(f"Error fetching message {message_id}: {e}")
+            raise FileNotFound(f"Message {message_id} not found")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)